source/cpu/pstate.go

```diff
@@ -32,12 +33,64 @@ func detectPstate() (map[string]string, error) {
 		return nil, nil
 	}
 
-	// Only looking for turbo boost for now...
+	// Get global pstate status
+	data, err := ioutil.ReadFile(source.SysfsDir.Path("devices/system/cpu/intel_pstate/status"))
+	if err != nil {
+		return nil, fmt.Errorf("could not read pstate status: %s", err.Error())
+	}
+	status := strings.TrimSpace(string(data))
+
+	// Check that intel_pstate driver is being used by looping through policies
+	policies, err := ioutil.ReadDir(source.SysfsDir.Path("devices/system/cpu/cpufreq"))
+	if err != nil {
+		return nil, fmt.Errorf("cpufreq directory not available to check pstate: %s", err.Error())
```

Comment:
The phrasing of this error message assumes the reason for the error returned from ReadDir(). Could the problem be related to permissions, or something else other than the directory not being there? Maybe here wrap the existing error with a "failed to read cpufreq..." general message and let the error from ReadDir report the reason.

Reply:
Done